lib/chef/provider/batch.rb

```diff
@@ -28,6 +28,14 @@ def initialize (new_resource, run_context)
         super(new_resource, run_context, '.bat')
       end
 
+      def command
+        basepath = is_forced_32bit ? wow64_directory : run_context.node.kernel.os_info.system_directory
+
+        interpreter_path = ::File.join(basepath, interpreter).gsub('/', '\\')
```

Comment:
Might be able to avoid the gsub by using one of the path helper functions like `Chef::Util::PathHelper.canonical_path` in https://github.com/chef/chef/blob/master/chef-config/lib/chef-config/path_helper.rb.